usd/include/sdf/usd/Conversions.hh

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+
+#ifndef SDF_USD_CONVERSIONS_HH_
+#define SDF_USD_CONVERSIONS_HH_
+
+#include <memory>
+
+#include <ignition/common/Material.hh>
+
+#include "sdf/Material.hh"
+#include "sdf/sdf_config.h"
+#include "sdf/usd/Export.hh"
+
+namespace sdf
+{
+  // Inline bracket to help doxygen filtering.
+  inline namespace SDF_VERSION_NAMESPACE {
+  //
+  namespace usd
+  {
+    /// \brief Specialized conversion from an Ignition Common Material
+    /// to a SDF material
+    /// \param[in] _in Ignition Common Material.
+    /// \return SDF material.
+    IGNITION_SDFORMAT_USD_VISIBLE
+    sdf::Material convert(
+        const std::shared_ptr<ignition::common::Material> &_in);
+
+    /// \brief Specialized conversion from an SDF material to a Ignition Common
+    /// material.
+    /// \param[in] _in SDF material.
+    /// \return Ignition Common Material.
+    IGNITION_SDFORMAT_USD_VISIBLE
+    std::shared_ptr<ignition::common::Material> convert(
```

Comment:
why return a `shared_ptr` here? Is it important that the resulting `Material` be allocated on the heap? we return an `sdf::Material` object from the other function

Reply:
not really sure about how to deal with this, because I can't return a `ignition::common::Material` because is a non-copyable class (it has a `unique_ptr` inside). I decided touse the same type in both methods. 

@scpeters what do you think is right type ?

Reply:
ok, that's reasonable. I'll open an issue in ign-common suggesting that we improve the Material class to make it copyable

Reply:
If it has a move ctor, then you should be might be able to return `ignition::common::Material` and the compiler will use copy elision to construct the object at the call location.

Reply:
it doesn't have a move ctor https://github.com/ignitionrobotics/ign-common/blob/ign-common4/graphics/include/ignition/common/Material.hh#L90-L97

@azeey, does it make sense to include it ? or can we keep the `shared_ptr` ?

Reply:
Unfortunately, since it has a user declared destructor, the compiler will not generate a move ctor. Simply removing the destructor would fix this, but I think that will break ABI. So, returning a smart pointer seems to be our best bet, but if we have to do that, my preference would be to use a `unique_ptr` unless there is a particular reason to use a `shared_ptr`.

Reply:
`SubMesh::MaterialByIndex` returns a `shared_ptr` which mean we need to convert this `shared_ptr` to `unique_ptr` to finally get the `sdf::Material`. I will bet for the `shared_ptr`to simplify things.

Reply:
I have an idea for getting around the `std::shared_ptr` issue - see https://github.com/ignitionrobotics/sdformat/pull/831/commits/3095a50c29269fa54a93db9d6355372a608fc4a2. If you guys think this isn't a good solution, feel free to revert this commit.